src/backend/main/go.main/contact/contact.go

```diff
@@ -0,0 +1,210 @@
+// Copyleft (ɔ) 2019 The Caliopen contributors.
+// Use of this source code is governed by a GNU AFFERO GENERAL PUBLIC
+// license (AGPL) that can be found in the LICENSE file.
+
+package contact
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"github.com/CaliOpen/Caliopen/src/backend/defs/go-objects"
+	"github.com/CaliOpen/Caliopen/src/backend/main/go.main/helpers"
+	log "github.com/Sirupsen/logrus"
+	"github.com/emersion/go-vcard"
+	"github.com/keybase/go-crypto/openpgp"
+	"github.com/satori/go.uuid"
+	"strings"
+)
+
+// Parse EMAIL vcard field
+func parseEmail(field *vcard.Field) *objects.EmailContact {
+	email := new(objects.EmailContact)
+	uid := new(objects.UUID)
+	_ = uid.UnmarshalBinary(uuid.NewV4().Bytes())
+	email.EmailId = *uid
+	email.Address = strings.ToLower(field.Value)
+	if field.Params["TYPE"] != nil {
+		email.Type = strings.ToLower(field.Params["TYPE"][0])
+	}
+	// TODO complete struct filling
+	email.Label = field.Value
+	return email
+}
+
+// Parse TEL vcard field
+func parsePhone(field *vcard.Field) *objects.Phone {
+	phone := new(objects.Phone)
+	uid := new(objects.UUID)
+	_ = uid.UnmarshalBinary(uuid.NewV4().Bytes())
+	phone.PhoneId = *uid
+	phone.Number = field.Value
+	if field.Params["TYPE"] != nil {
+		phone.Type = strings.ToLower(field.Params["TYPE"][0])
+	}
+	return phone
+}
+
+// Parse IMPP vcard field
+func parseIm(field *vcard.Field) *objects.IM {
+	im := new(objects.IM)
+	uid := new(objects.UUID)
+	_ = uid.UnmarshalBinary(uuid.NewV4().Bytes())
+	im.IMId = *uid
+	im.Address = strings.ToLower(field.Value)
+	im.Label = field.Value
+	if field.Params["TYPE"] != nil {
+		im.Type = strings.ToLower(field.Params["TYPE"][0])
+	}
+	return im
+}
+
+// Read an armored PGP public key and return an openpgp.Entity structure
+func readPgpKey(pubkey []byte) (*openpgp.Entity, error) {
+	reader := bytes.NewReader(pubkey)
+	var entitiesList openpgp.EntityList
+	var err error
+
+	entitiesList, err = openpgp.ReadArmoredKeyRing(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	//handle only first key found for now
+	if len(entitiesList) > 1 {
+		return nil, errors.New("More than one key found in payload")
+	}
+	return entitiesList[0], nil
+}
+
+// Parse KEY vcard field and transform to a PublicKey that belong to a contact
+func parseKey(field *vcard.Field, contact *objects.Contact) (*objects.PublicKey, error) {
```

Comment:
elsewhere in our stack, we use to return a nil pointer if func fails, instead of returning an empty or incomplete object. Here var `key` is return even if there were an error.

Reply:
done in #12bbebf81